### PR TITLE
added the ldap authentication.

### PR DESCRIPTION
**Pull request by aoyawale**

### CODE REVIEW

File: content/doc/features/authentication.en-us.md [hunk 1]
```diff
@@ -13,6 +13,180 @@ menu:
...
+
+- Skip TLS Verify
+  - Disable TLS verify on authentication.
+  
+- This authentication is activate
+  - Enable or disable this auth.
+
+## Freeipa
+
+- In order to login to the Gogs using FreeIPA credentials, you need to create a bind account for Gogs to use:
```
Comment (lunny) [inline content/doc/features/authentication.en-us.md]: Gogs should be Gitea.

**Comment by lunny** [comment]
We also support OAuth(google+, github, twitter and about 20+ services) and OpenID login, if you can add them, that's be better. Of course, currently is also OK except some word `Gogs` should be replaced by `Gitea`.

File: content/doc/features/authentication.en-us.md (hunk 1 above)
Comment (aoyawale) [reply content/doc/features/authentication.en-us.md]: changed the doc part to Gitea but, are the config files in this doc still point to gogs or should be changed to gitea too?

**Comment by lafriks** [comment]
Please also add LDAP sample for Microsoft AD:

Port: `389`
User search base: `(&(objectCategory=Person)(memberOf=CN=user-group,OU=example,DC=example,DC=org)(sAMAccountName=%s)(!(UserAccountControl:1.2.840.113556.1.4.803:=2)))`
Admin filter: `(memberOf=CN=admin-group,OU=example,DC=example,DC=org)`
Username: `sAMAccountName`

File: content/doc/features/authentication.en-us.md [hunk 2]
```diff
@@ -13,6 +13,191 @@ menu:
...
+- Enable TLS Encryption
+  - Enable TLS encryption on authentication.
+
+- Skip TLS Verify
+  - Disable TLS verify on authentication.
+  
+- This authentication is activate
+  - Enable or disable this auth.
+
+## Microsoft AD
```
Comment (lafriks) [inline content/doc/features/authentication.en-us.md]: I was thinking that these need to be added under LDAP section as additional example lines for appropriate fields

**Review by lafriks — CHANGES_REQUESTED** [review]

File: content/doc/features/authentication.en-us.md (hunk 2 above)
Comment (aoyawale) [reply content/doc/features/authentication.en-us.md]: oops sorry for that. Let me change it.

**Comment by aoyawale** [comment]
I can update for the other authentication we support but, I don't have docs for those.

**Comment by lunny** [comment]
@jlozadad OK. that should be another PR.

**Comment by lunny** [comment]
LGTM

**Comment by strk** [comment]
There are references to "Gogs" still, in its current version, can you update them ?
Also, it would be great to describe what `Verify group membership in LDAP` does, exactly.

**Comment by aoyawale** [comment]
looks like my previous comment got buried with the commit. But, yes I was asking the same thing. Some config still shows "gogs" and I was wondering if that part changed in gitea too or still the same. If all of it changed to gitea then plz confirm and I'll change it again.

**Comment by lafriks** [comment]
@jlozadad I was thinking that these examples needs to be added not in separate section but at the end of each description as additional example.
For example:
```
- User Search Base **(required)**
  - The LDAP base at which user accounts will be searched for.
  - Example: `ou=Users,dc=mydomain,dc=com`
  - Example for Microsoft AD: `(&(objectCategory=Person)(memberOf=CN=user-group,OU=example,DC=example,DC=org)(sAMAccountName=%s)(!(UserAccountControl:1.2.840.113556.1.4.803:=2)))`
```
etc

**Comment by aoyawale** [comment]
so basically copy the example you mentioned and add it to the others? I'm not sure what would look different for the others.

**Comment by lafriks** [comment]
@jlozadad yes add examples for Microsoft AD to sections port, user search base, admin filter and username

**Comment by aoyawale** [comment]
I added the changes. Let me know if its correct.

**Comment by lafriks** [comment]
Not really, I did mean only parts that are meant for each section like I did give example in comment 5 days ago

**Comment by appleboy** [comment]
any updates on this?

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
Fixed

**Comment by daviian** [comment]
LGTM

**Pull request merged**